### ISSUE TITLE
Startup fails with exception

### ISSUE THREAD

**Issue by timlinux** [qgis/homebrew-qgisdev#27]
Hi @dakcarto 

So I updated with your new dependencies provided today. I get a crash on startup.
```
src/core/qgsnetworkaccessmanager.cpp: 164: (setFallbackProxyAndExcludes) [11ms] proxy settings: (type:DefaultProxy host: :0, user:, password:not set
src/core/qgsnetworkaccessmanager.cpp: 381: (setupDefaultProxyAndCache) [3ms] cacheDirectory: /Users/timlinux/.qgis3/cache/
src/core/qgsnetworkaccessmanager.cpp: 382: (setupDefaultProxyAndCache) [0ms] maximumCacheSize: 0
2017-03-15 07:24:39.832 QGIS[95829:12850174] *** Assertion failure in -[NSBitmapImageRep initWithCGImage:], /Library/Caches/com.apple.xbs/Sources/AppKit/AppKit-1504.81.100/AppKit.subproj/NSBitmapImageRep.m:1296
src/core/qgsapplication.cpp: 324: (notify) [643ms] Caught unhandled unknown exception
2017-03-15 07:24:41.211 QGIS[95829:12850174] *** Assertion failure in -[NSBitmapImageRep initWithCGImage:], /Library/Caches/com.apple.xbs/Sources/AppKit/AppKit-1504.81.100/AppKit.subproj/NSBitmapImageRep.m:1296
src/core/qgsapplication.cpp: 324: (notify) [1380ms] Caught unhandled unknown exception
src/core/qgsruntimeprofiler.cpp: 47: (end) [987ms] PROFILE: qgisapp/startup/Setting up UI - 3.01
src/core/qgsruntimeprofiler.cpp: 47: (end) [22ms] PROFILE: qgisapp/startup/Checking database - 0.022
src/core/auth/qgsauthmanager.cpp: 106: (init) [5ms] Initializing QCA...
src/core/auth/qgsauthmanager.cpp: 109: (init) [0ms] QCA initialized.
```

**Comment by timlinux** [comment]
I'll add some more diagnostics later today

**Comment by m-kuhn** [comment]
For reference:

https://bugreports.qt.io/browse/QTBUG-57706
https://bugreports.qt.io/browse/QTBUG-58344

Not sure if I interpret these resources correctly. But it looks to be a bug in Qt 5.8 and fixed in 5.8.1.

**Comment by timlinux** [comment]
I'm still building here under 5.7 and the problem is magically fixed so I am not sure what the cause was...

**Comment by m-kuhn** [comment]
Was that ever a problem under 5.7?

**Comment by dakcarto** [comment]
Startup exceptions fixed in https://github.com/qgis/QGIS/commit/d303ca8

The Qt 5.8.0 regression remains. From QTBUG-58344, I surmised that the exception was caused by a missing QIcon file reference when QgisApp's QMainWindow GUI was loading. I found two such old PNG references in `qgisapp.ui`. Until the regression is fixed in Qt 5.8.x, the exceptions may still be thrown in similar circumstances from other parts of the app or with third-party plugins.
